pandas/tests/series/test_replace.py

```diff
@@ -292,6 +292,23 @@ def test_replace_categorical(self, categorical, numeric):
         expected = pd.Series(numeric)
         tm.assert_series_equal(expected, result, check_dtype=False)
 
+    def test_replace_categorical_single(self):
+        # GH 26988
```

Comment:
the `copy` semantics of Categorical are weird (#26995).  Can you check that when you do `__setitem__` below it actually alters the appropriate objects (and only the appropriate ones)?

Reply:
I can confirm that after `__setitem__`, `expected` has its second element changed but `c` doesn't, as intended